packages/commercetools/theme/pages/Checkout/Shipping.vue

```diff
@@ -211,39 +251,120 @@ export default {
       loadDetails,
       loading
     } = useCheckout();
+    const { addresses: shippingAddresses, load: loadShippingAddresses, setDefault } = useUserShipping();
+    const { isAuthenticated } = useUser();
+
+    const provideAddress = ref(true);
+    const justChangedAddress = ref(false);
+    const currentAddressId = ref(-1);
+    const setAsDefault = ref(false);
+
+    const findAddressById = id => shippingAddresses.value.find(address => address.id === id);
+
+    const setCurrentAddress = async (addressId) => {
+      const chosenAddress = findAddressById(addressId);
+      if (!chosenAddress) {
+        return;
+      }
+      currentAddressId.value = addressId;
+      setShippingDetails({
+        ...shippingDetails.value,
+        contactInfo: {
+          ...shippingDetails.value.contactInfo,
+          phone: chosenAddress.phoneNumber
+        },
+        streetNumber: chosenAddress.apartment,
```

Comment:
why do you need such a mapping in the theme? you use pure CT interfaces can't you just keep that convention? I think it's possible to have just one line eg `setShippingDetails(chosenAddress)`